advanced_source/super_resolution_with_onnxruntime.py

```diff
@@ -0,0 +1,285 @@
+
+``NOTE``: This tutorial needs PyTorch master branch which can be installed by following
+the instructions `here <https://github.com/pytorch/pytorch#from-source>`__
+
+"""
+
+# Some standard imports
+import io
+import numpy as np
+
+from torch import nn
+import torch.utils.model_zoo as model_zoo
+import torch.onnx
+
+
+######################################################################
+# Super-resolution is a way of increasing the resolution of images, videos
+# and is widely used in image processing or video editing. For this
+# tutorial, we will first use a small super-resolution model with a dummy
+# input.
+#
+# First, let's create a SuperResolution model in PyTorch. `This
+# model <https://github.com/pytorch/examples/blob/master/super_resolution/model.py>`__
+# comes directly from PyTorch's examples without modification:
+#
+
+# Super Resolution model definition in PyTorch
+import torch.nn as nn
+import torch.nn.init as init
+
+
+class SuperResolutionNet(nn.Module):
+    def __init__(self, upscale_factor, inplace=False):
+        super(SuperResolutionNet, self).__init__()
+
+        self.relu = nn.ReLU(inplace=inplace)
+        self.conv1 = nn.Conv2d(1, 64, (5, 5), (1, 1), (2, 2))
+        self.conv2 = nn.Conv2d(64, 64, (3, 3), (1, 1), (1, 1))
+        self.conv3 = nn.Conv2d(64, 32, (3, 3), (1, 1), (1, 1))
+        self.conv4 = nn.Conv2d(32, upscale_factor ** 2, (3, 3), (1, 1), (1, 1))
+        self.pixel_shuffle = nn.PixelShuffle(upscale_factor)
+
+        self._initialize_weights()
+
+    def forward(self, x):
+        x = self.relu(self.conv1(x))
+        x = self.relu(self.conv2(x))
+        x = self.relu(self.conv3(x))
+        x = self.pixel_shuffle(self.conv4(x))
+        return x
+
+    def _initialize_weights(self):
+        init.orthogonal_(self.conv1.weight, init.calculate_gain('relu'))
+        init.orthogonal_(self.conv2.weight, init.calculate_gain('relu'))
+        init.orthogonal_(self.conv3.weight, init.calculate_gain('relu'))
+        init.orthogonal_(self.conv4.weight)
+
+# Create the super-resolution model by using the above model definition.
+torch_model = SuperResolutionNet(upscale_factor=3)
+
+
+######################################################################
+# Ordinarily, you would now train this model; however, for this tutorial,
+# we will instead download some pre-trained weights. Note that this model
+# was not trained fully for good accuracy and is used here for
+# demonstration purposes only.
+#
+
+# Load pretrained model weights
+model_url = 'https://s3.amazonaws.com/pytorch/test_data/export/superres_epoch100-44c6958e.pth'
+batch_size = 1    # just a random number
+
+# Initialize model with the pretrained weights
+map_location = lambda storage, loc: storage
+if torch.cuda.is_available():
+    map_location = None
+torch_model.load_state_dict(model_zoo.load_url(model_url, map_location=map_location))
+
+# set the train mode to false since we will only run the forward pass.
+torch_model.train(False)
```

Comment:
you can also mention torch_model.eval() has the same affect as torch_model.train(False)

Reply:
Also explain why calling this method is important, because it will turn model to inference mode. Otherwise, ops like dropout, batchnorm may behave differently.

Reply:
good point, I didn't notice this comment before.
Adding to the next commit.